Composer/packages/form-dialogs/src/components/property/PropertyTypeSelector.tsx

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { useId } from '@uifabric/react-hooks';
+import formatMessage from 'format-message';
+import { Dropdown, DropdownMenuItemType, IDropdownOption } from 'office-ui-fabric-react/lib/Dropdown';
+import * as React from 'react';
+import { useRecoilValue } from 'recoil';
+import { formDialogTemplatesAtom } from 'src/atoms/appState';
+import {
+  ArrayPropertyPayload,
+  BuiltInStringFormat,
+  builtInStringFormats,
+  FormDialogPropertyPayload,
+  IntegerPropertyPayload,
+  NumberPropertyPayload,
+  RefPropertyPayload,
+  FormDialogPropertyKind,
+  StringPropertyPayload,
+} from 'src/atoms/types';
+import { FieldLabel } from 'src/components/common/FieldLabel';
+
+const processSelection = (
+  isArray: boolean,
+  oldKind: FormDialogPropertyKind,
+  newKind: FormDialogPropertyKind,
+  selectedKey: string,
+  payload: FormDialogPropertyPayload
+): FormDialogPropertyPayload => {
+  if (isArray) {
+    return processSelection(
+      false,
+      (payload as ArrayPropertyPayload).items.kind,
+      newKind,
+      selectedKey,
+      (payload as ArrayPropertyPayload).items
+    );
+  }
+
+  if (oldKind === newKind) {
```

Comment:
I think a comment is needed here to understand what happens when oldKind === newKind. Most code that has this comparison is optimizing to a no-op when there isn't a change, but you return a different result than when there is a change.